### ISSUE TITLE
Helm chart creates clusters with insufficient permissions if the release is not named `zalando`

### ISSUE THREAD

**Issue by teozkr** [zalando/postgres-operator#528]
The chart creates the ClusterRole using the following configuration:

https://github.com/zalando/postgres-operator/blob/c243074083d7cb15eeb17e9ec94899bc51ac37f4/charts/postgres-operator/templates/clusterrole.yaml#L1-L5 

This creates a ClusterRole named `{{ .Release.Name }}-{{ .Chart.Name }}` (`{{ .Release.Name }}-postgres-operator` if you're not using aliases). However, Postgres-Operator always tries to assign the created ServiceAccount to the ClusterRole `zalando-postgres-operator`. If the Helm release wasn't named `zalando` then this leaves the pods' serviceaccount without the correct permissions.

When this happens then Postgres-Operator logs the following warning:

>     time="2019-04-01T13:36:57Z" level=warning msg="Pods and/or Patroni may misfunction due to the lack of permissions: could not create role binding zalando-postgres-operator : cannot bind the pod service account \"etimo-postgres-operator\" defined in the config map to the cluster role in the \"default\" namespace: clusterroles.rbac.authorization.k8s.io \"zalando-postgres-operator\" not found" pkg=controller

While the pods keep spamming the following error:

> 
> ```
> 2019-04-01 14:05:13,937 WARNING: Can not get cluster from dcs
> 2019-04-01 14:05:13,942 ERROR: get_cluster
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/dist-packages/patroni/dcs/kubernetes.py", line 140, in _load_cluster
>     response = self.retry(self._api.list_namespaced_pod, self._namespace, label_selector=self._label_selector)
>   File "/usr/local/lib/python3.6/dist-packages/patroni/dcs/kubernetes.py", line 112, in retry
>     return self._retry.copy()(*args, **kwargs)
>   File "/usr/local/lib/python3.6/dist-packages/patroni/utils.py", line 254, in __call__
>     return func(*args, **kwargs)
>   File "/usr/local/lib/python3.6/dist-packages/patroni/dcs/kubernetes.py", line 50, in wrapper
>     return getattr(self._api, func)(*args, **kwargs)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/apis/core_v1_api.py", line 12310, in list_namespaced_pod
>     (data) = self.list_namespaced_pod_with_http_info(namespace, **kwargs)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/apis/core_v1_api.py", line 12413, in list_namespaced_pod_with_http_info
>     collection_formats=collection_formats)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/api_client.py", line 321, in call_api
>     _return_http_data_only, collection_formats, _preload_content, _request_timeout)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/api_client.py", line 155, in __call_api
>     _request_timeout=_request_timeout)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/api_client.py", line 342, in request
>     headers=headers)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/rest.py", line 231, in GET
>     query_params=query_params)
>   File "/usr/local/lib/python3.6/dist-packages/kubernetes/client/rest.py", line 222, in request
>     raise ApiException(http_resp=r)
> kubernetes.client.rest.ApiException: (403)
> Reason: Forbidden
> HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Date': 'Mon, 01 Apr 2019 14:05:13 GMT', 'Content-Length': '298'})
> HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"system:serviceaccount:default:etimo-postgres-operator\" cannot list resource \"pods\" in API group \"\" in the namespace \"default\"","reason":"Forbidden","details":{"kind":"pods"},"code":403}
> ```

**Comment by teozkr** [comment]
This seems to be hard-coded here:

https://github.com/zalando/postgres-operator/blob/c243074083d7cb15eeb17e9ec94899bc51ac37f4/pkg/controller/controller.go#L194-L210

**Comment by teozkr** [comment]
As a workaround you can override the role binding from your `values.yaml`:

```
# Workaround for https://github.com/zalando/postgres-operator/issues/528
postgres-operator:
  config:
    pod_service_account_role_binding_definition: |
      apiVersion: rbac.authorization.k8s.io/v1beta1
      kind: RoleBinding
      metadata:
        name: etimo-postgres-operator
      roleRef:
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
        name: etimo-postgres-operator
      subjects:
      - kind: ServiceAccount
        name: operator
```

However, this is still bound to a specific Helm release name, since templates aren't available in `values.yaml`.

**Comment by sdudoladov** [comment]
@FxKu pls have  a look

**Comment by FxKu** [comment]
Could reproduce it. Will fix, but for now, I think we should update the docs to `helm install --name zalando ...`

**Comment by sdudoladov** [comment]
closed by #529
